tests/integration/test_pagure.py

```diff
@@ -19,54 +19,79 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 import os
 import subprocess
+from datetime import datetime
 from pathlib import Path
 
-import pytest
 from ogr.services.pagure import PagureService
+from vcr import use_cassette
 
 from tests.spellbook import git_set_user_email
 
 
-@pytest.mark.skipif(
-    condition=True, reason="Don't interact with a real pagure instance by default"
-)
+RESPONSE_HEADERS_TO_DROP = [
+    "X-Fedora-RequestID",
+    "content-security-policy",
+    "set-cookie",
+]
+
+
+def drop_sensitive_response_data(response):
+    for h in RESPONSE_HEADERS_TO_DROP:
+        try:
+            del response["headers"][h]
+        except KeyError:
+            pass
+    return response
+
+
 def test_basic_distgit_workflow(tmpdir):
-    pagure_token = os.getenv("PAGURE_TOKEN")
+    # if pagure_token is not set and we're recording, we'll get a token failure,
+    # but that's expected so no need to take care of it
+    with use_cassette(
+        path=str(Path(__file__).parent / "test_basic_distgit_workflow"),
```

Comment:
We still need to use the logic from requre to have one file per test. Or some unittest super class to handle that.

Reply:
OK, there is some support for this:
- https://vcrpy.readthedocs.io/en/latest/advanced.html#automatic-cassette-naming